esp-mbedtls/src/lib.rs

```diff
@@ -87,10 +88,83 @@ pub fn set_debug(level: u32) {
     }
 }
 
+/// Holds a X509 certificate
+///
+/// # Examples
+/// Initialize with a PEM certificate
+/// ```
+/// const CERTIFICATE: &[u8] = include_bytes!("certificate.pem");
+/// let cert = X509::pem(CERTIFICATE).unwrap();
+/// ```
+///
+/// Initialize with a DER certificate
+/// ```
+/// const CERTIFICATE: &[u8] = include_bytes!("certificate.der");
+/// let cert = X509::der(CERTIFICATE);
+/// ```
+#[derive(Copy, Clone, Eq, PartialEq)]
+pub struct X509<'a> {
+    /// The certificate in bytes
+    bytes: &'a [u8],
+
+    /// The length of the certificate
+    len: usize,
```

Comment:
maybe I missed something but the `bytes` slice already knows it's length so we don't need to store the length here

Reply:
This is for a specific use case where bytes would be a `[0u8; T]`, with extra 0s at the end, the certificate not taking the full size of it.